src/rinoh/util/rkd.py

```diff
@@ -0,0 +1,31 @@
+from collections.abc import MutableMapping
+from weakref import ref
+
+
+
+# http://stackoverflow.com/a/3387975/438249
+class RefKeyDictionary(MutableMapping):
+    """A dictionary that compares keys based on their id (address). Hence, the
+    keys can be mutable."""
+
+    def __init__(self, *args, **kwargs):
+        self.store = dict()
+        self.update(dict(*args, **kwargs))
```

Comment:
Why not just `self.update(*args, **kwargs)`?

Actually, it does not work with kwargs at all, because you create a weak reference to a key, but kwargs keys are strings, which do not support weak references.

Reply:
Because I copied this code from StackOverflow without thinking :-)